### DISCUSSION TITLE
out of memory

### DISCUSSION THREAD

**Discussion by orion-tan** [matt-42/lithium#78] [Q&A]
When I first tried it on a 1-core 2G ubuntu 20 using
`sudo li run myfile`
I got an out-of-memory signal to stop compiling at 50%, but when I created a 2G swap space and compiled at 54%, I'd like to know how much memory is needed for the first compilation?

the eroor just like this:  Killed signal terminated program cc1plus
compilation terminated.

**Comment by matt-42** [comment]
Hi @ToxicCloud 

What program are you trying to compile ? Could you run the hello_world example on the same config ? try this one:  https://github.com/matt-42/lithium/blob/master/libraries/http_server/tests/hello_world.cc (on my pc it takes about 400MB of ram to compile).
